src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int POSITION_FOR_NULL_KEYS = 0;
+    private int size;
+    private int mapCapacity;
+    private Node<K, V>[] table;
+    private int threshold;
+
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            table = resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (key == null) {
+            if (table[POSITION_FOR_NULL_KEYS] == null) {
+                table[POSITION_FOR_NULL_KEYS] = newNode;
+                size++;
+            } else {
+                Node<K, V> node = table[POSITION_FOR_NULL_KEYS];
+                if (node.key == null) {
+                    node.value = value;
+                    return;
+                }
+                while (node.next != null) {
+                    if (node.next.key == null) {
+                        node.next.value = value;
+                        return;
+                    }
+                    node = node.next;
+                }
+                node.next = newNode;
+                size++;
+            }
+        } else {
+            Node<K, V> node = table[hash(key)];
+            if (node == null) {
+                table[hash(key)] = newNode;
+                size++;
+            } else if (node.key.equals(key)) {
+                node.value = value;
+            } else {
+                while (node.next != null) {
+                    if (node.next.key != null && node.next.key.equals(key)) {
+                        node.next.value = value;
+                        return;
+                    } else {
+                        node = node.next;
+                    }
+                }
+                node.next = newNode;
+                size++;
+            }
+        }
+        if (size + 1 > threshold) {
+            table = resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+        for (Node<K, V> node : table) {
```

Comment:
we could find backet index by key hash, why do we need go through all buckets in table?